File: Maintenance/LSM-9506-ESP32_Send.py
#include <WiFi.h>
#include <WiFiClient.h>

const char* ssid = "WescanRichmond";
const char* password = "PYL0Ne1ab";

const char* serverIP = "192.168.1.100"; // Replace with your computer's IP address
const int serverPort = 5555; // Choose a port number

WiFiClient client;

void setup() {
  Serial.begin(115200);
  delay(1000);

  Serial.println("Connecting to WiFi...");
  WiFi.begin(ssid, password);
  
  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.println("Connecting...");
  }
  Serial.println("Connected to WiFi");
}

void loop() {
  if (!client.connected()) {
    Serial.println("Connecting to server...");
    if (client.connect(serverIP, serverPort)) {
      Serial.println("Connected to server");
      client.println("Hello from ESP32"); // Replace with the data you want to send
    } else {
      Serial.println("Connection failed");
    }
  }

  delay(5000); // Adjust the delay as needed
}

File: Maintenance/LSM-9506-ESP32_Send.cpp
// ESP32 code to handle RS232 communication and Wi-Fi connectivity
// Implement RS232 communication and Wi-Fi setup
// Pseudocode for handling Wi-Fi commands and RS232 data forwarding

#include <WiFi.h>
#include <HardwareSerial.h>

const char* ssid = "WescanRichmond";
const char* password = "PYL0Ne1ab";
WiFiServer server(80); // Change port number if needed

HardwareSerial rs232(2); // Configure RS232 (UART) communication

bool LSM_Read = false; // Flag to trigger reading from RS232
bool Wait = false; // Flag to trigger waiting for RS232 data

void loop() {
    WiFiClient client = server.available();
    if (client) {
        while (client.connected()) {
            if (client.available()) {
                String command = client.readStringUntil('\r');
                if (command == "START_READING") {
                    LSM_Read = true; // Set flag to start reading from RS232
                    client.println("Reading started"); // Send acknowledgment to the computer
                }
                String command client.readStringUntil('\r');
                if (command = "STOP_READING") {
                    LSM_Read = false; // Set flag to stop reading from RS232
                    client.println("Reading stopped"); // Send acknowledgment to the computer
                
                // Handle other commands if needed
                }
            }
        }
        client.stop();
    }

    // When the flag is set, start reading data from RS232
    while (LSM_Read) {
        // Read data from RS232 and process it
      StringData = rs232.readStringUntil('\r');
      if (rs232Data.startswith = 'ER')
        if (rs232Data = 'ER0') {
          Wait = false;   // reset wait flag
          err = 0;    // reset error counter
        }
        else err++;    // increment error counter
      else {
        if (wait = false) {
          // Send data to computer
          Wait = true;    // set wait flag
          err = 0;    // reset error counter
          client.print(rs232Data);
          }
        else err = 0;    // reset error counter
      }
    }
}